src/base/bittorrent/session.cpp

```diff
@@ -1405,9 +1405,14 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
 
     // Remove it from session
     if (deleteLocalFiles) {
-        QString rootPath = torrent->rootPath(true);
-        if (!rootPath.isEmpty())
-            m_savePathsToRemove[torrent->hash()] = rootPath;
+        if (torrent->savePath(true) == torrentTempPath(torrent->hash())) {
+            m_savePathsToRemove[torrent->hash()] = torrent->savePath(true);
+        }
+        else {
+            QString rootPath = torrent->rootPath(true);
+            if (!rootPath.isEmpty())
+                m_savePathsToRemove[torrent->hash()] = rootPath;
+        }
         m_nativeSession->remove_torrent(torrent->nativeHandle(), libt::session::delete_files);
     }
     else {
```

Comment:
> but many users misunderstand its meaning (temp folder) 

Precisely! And since this fix is for them, it is important to take into account that in the current context these are *not* temp dirs.

> We need to figure this out definitely.

Although I never used temp directory, but may I suggest to apply hidden attribute to it and maybe its contents too, and hope that this will speak for itself, or in the worst case render those directories useless for user files.